pkg/logger/logger.go

```diff
@@ -103,51 +113,230 @@ func (l *Logger) log(logColor *color.Color, message string) {
+		if level >= log.ErrorLevel && len(atmosErr.Tips) > 0 {
+			PrintTips(atmosErr.Tips)
+		}
+	}
+}
+
+// Log implements the core logging functionality with intelligent handling of different input types.
+func (l *AtmosLogger) Log(level log.Level, msgOrErr interface{}, keyvals ...interface{}) {
+	// Process the message based on type
+	msg, processedKeyvals := processLogMessage(msgOrErr, keyvals)
+
+	// Validate key-value pairs
+	processedKeyvals = validateKeyValuePairs(l, processedKeyvals)
+
+	// Log the message
+	l.Logger.Log(level, msg, processedKeyvals...)
+
+	// Handle tips for display
+	handleTips(level, msgOrErr)
+}
+
+// Error logs at ERROR level with different input types.
+// It accepts:
+// - String message: Error("event_name", "key", "value").
+// - Error object: Error(err, "key", "value").
+// - AtmosError: Error(atmosErr, "key", "value"). // atmosErr is of type *errors.AtmosError.
+func (l *AtmosLogger) Error(msgOrErr interface{}, keyvals ...interface{}) {
+	l.Log(log.ErrorLevel, msgOrErr, keyvals...)
+}
+
+// Warning logs at WARN level with different input types.
+// It accepts:
+// - String message: Warning("event_name", "key", "value").
+// - Error object: Warning(err, "key", "value").
+// - AtmosError: Warning(atmosErr, "key", "value"). // atmosErr is of type *errors.AtmosError.
+func (l *AtmosLogger) Warning(msgOrErr interface{}, keyvals ...interface{}) {
+	l.Log(log.WarnLevel, msgOrErr, keyvals...)
+}
+
+// Info logs at INFO level with different input types.
+// It accepts:
+// - String message: Info("event_name", "key", "value").
+// - Error object: Info(err, "key", "value").
+// - AtmosError: Info(atmosErr, "key", "value"). // atmosErr is of type *errors.AtmosError.
+func (l *AtmosLogger) Info(msgOrErr interface{}, keyvals ...interface{}) {
+	l.Log(log.InfoLevel, msgOrErr, keyvals...)
+}
+
+// Debug logs at DEBUG level with different input types.
+// It accepts:
+// - String message: Debug("event_name", "key", "value").
+// - Error object: Debug(err, "key", "value").
+// - AtmosError: Debug(atmosErr, "key", "value"). // atmosErr is of type *errors.AtmosError.
+func (l *AtmosLogger) Debug(msgOrErr interface{}, keyvals ...interface{}) {
+	l.Log(log.DebugLevel, msgOrErr, keyvals...)
+}
+
+// Note: Charmbracelet logger doesn't have a TraceLevel, so we need to implement a custom method in another PR.
+
+// SetAtmosLogLevel configures the logger level based on Atmos log level strings.
+func SetAtmosLogLevel(l *AtmosLogger, level string) error {
+	_, err := ParseLogLevel(level)
+	if err != nil {
+		return err
+	}
+
+	charmLevel, exists := atmosToCharmLevelMap[level]
+	if exists {
+		l.SetLevel(charmLevel)
+	}
+	return nil
+}
+
+// PrintTips displays user guidance tips in a console-friendly format.
+func PrintTips(tips []string) {
+	if len(tips) == 0 {
+		return
+	}
+
+	var output io.Writer = os.Stderr
+	if color.Error != nil {
+		output = color.Error
+	}
+
+	for i, tip := range tips {
+		fmt.Fprintf(output, "  %s %s\n",
```

Comment:
I believe we have a new function for printing console messages, that ensures they always go to stderr.